baseclass/src/main/java/com/google/apigee/calloutbaseclass/CalloutsBase.java

```diff
@@ -0,0 +1,207 @@
+package com.google.apigee.calloutbaseclass;
+
+import com.apigee.flow.message.MessageContext;
+
+import java.io.PrintWriter;
+import java.io.StringWriter;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+/**
+ * Copyright 2020 Google LLC
+ *
+ * <p>Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file
+ * except in compliance with the License. You may obtain a copy of the License at
+ *
+ * <p>https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * <p>Unless required by applicable law or agreed to in writing, software distributed under the
+ * License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * <p>Base Class for Apigee Java Callouts containing commonly used methods across callouts.
+ */
+public abstract class CalloutsBase {
+
+  private static final String commonErrorPatternString = "^(.+?)[:;] (.+)$";
+  private static final String variableReferencePatternString =
+      "(.*?)\\{([^\\{\\} :][^\\{\\} ]*?)\\}(.*?)";
+
+  private static final String ERROR_FLOW_VARIABLE = "error";
+  private static final String EXCEPTION_FLOW_VARIABLE = "exception";
+  private static final String EXCEPTION_STACKTRACE_FLOW_VARIABLE = "exception_stacktrace";
+  private static final String LOG_FLOW_VARIABLE = "log";
+
+  private static final Pattern commonErrorPattern = Pattern.compile(commonErrorPatternString);
+  private static final Pattern variableReferencePattern =
+      Pattern.compile(variableReferencePatternString);
+
+  private final Map<String, String> properties;
```

Comment:
Same as above:

It seems the Properties class (https://docs.oracle.com/javase/7/docs/api/java/util/Properties.html) is meant for reading/writing into properties files and has special methods to do that. It's also implemented as just a Hashtable so I don't see any benefits of using the Properties class instead of a Map.